### PR TITLE
Mapping elements

### PR DESCRIPTION
**Pull request by HajimeKawahara**

### CODE REVIEW

**Comment by HajimeKawahara** [comment]
Now, we are ready for review. 
Check 
`python examples/mapping_examples.py`
it should generate like 
map


and unittest
`pytest`

**Comment by xr0038** [comment]
`pytest` returns the following error:

```
    def test_inout_detector():
        import pkg_resources
        from telescope_baseline.mapping.read_catalog import read_jasmine_targets
        hdf=pkg_resources.resource_filename('telescope_baseline', 'data/cat.hdf')
>       targets,l,b=read_jasmine_targets(hdf)

mapping/test_aperture.py:55: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
../../../.pyenv/versions/anaconda3-2020.11/lib/python3.8/site-packages/telescope_baseline-0.1.0-py3.8.egg/telescope_baseline/mapping/read_catalog.py:40: in read_jasmine_targets
    c = SkyCoord(ra=ra*u.degree, dec=dec*u.degree, frame='icrs')
../../../.pyenv/versions/anaconda3-2020.11/lib/python3.8/site-packages/pandas/core/ops/common.py:65: in new_method
    return method(self, other)
../../../.pyenv/versions/anaconda3-2020.11/lib/python3.8/site-packages/pandas/core/ops/__init__.py:343: in wrapper
    result = arithmetic_op(lvalues, rvalues, op)
../../../.pyenv/versions/anaconda3-2020.11/lib/python3.8/site-packages/pandas/core/ops/array_ops.py:190: in arithmetic_op
    res_values = na_arithmetic_op(lvalues, rvalues, op)
../../../.pyenv/versions/anaconda3-2020.11/lib/python3.8/site-packages/pandas/core/ops/array_ops.py:143: in na_arithmetic_op
    result = expressions.evaluate(op, left, right)
../../../.pyenv/versions/anaconda3-2020.11/lib/python3.8/site-packages/pandas/core/computation/expressions.py:233: in evaluate
    return _evaluate(op, op_str, a, b)  # type: ignore
../../../.pyenv/versions/anaconda3-2020.11/lib/python3.8/site-packages/pandas/core/computation/expressions.py:109: in _evaluate_numexpr
    result = ne.evaluate(
../../../.pyenv/versions/anaconda3-2020.11/lib/python3.8/site-packages/numexpr/necompiler.py:821: in evaluate
    signature = [(name, getType(arg)) for (name, arg) in
../../../.pyenv/versions/anaconda3-2020.11/lib/python3.8/site-packages/numexpr/necompiler.py:821: in <listcomp>
    signature = [(name, getType(arg)) for (name, arg) in
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

a = array(Unit("deg"), dtype=object)

    def getType(a):
        kind = a.dtype.kind
        if kind == 'b':
            return bool
        if kind in 'iu':
            if a.dtype.itemsize > 4:
                return long_  # ``long`` is for integers of more than 32 bits
            if kind == 'u' and a.dtype.itemsize == 4:
                return long_  # use ``long`` here as an ``int`` is not enough
            return int_
        if kind == 'f':
            if a.dtype.itemsize > 4:
                return double  # ``double`` is for floats of more than 32 bits
            return float
        if kind == 'c':
            return complex
        if kind == 'S':
            return bytes
>       raise ValueError("unknown type %s" % a.dtype.name)
E       ValueError: unknown type object

../../../.pyenv/versions/anaconda3-2020.11/lib/python3.8/site-packages/numexpr/necompiler.py:703: ValueError


=========================================== short test summary info ===========================================
FAILED mapping/test_aperture.py::test_inout_detector - ValueError: unknown type object
=================================== 1 failed, 7 passed, 6 warnings in 7.40s ===================================
```

**Comment by xr0038** [comment]
`python example/mapping_elements.py` also failed.

```
$ python examples/mapping_example.py 
Traceback (most recent call last):
  File "examples/mapping_example.py", line 20, in <module>
    targets,l,b=read_jasmine_targets(hdf)
  File "/home/ohsawa/.pyenv/versions/anaconda3-2020.11/lib/python3.8/site-packages/telescope_baseline-0.1.0-py3.8.egg/telescope_baseline/mapping/read_catalog.py", line 40, in read_jasmine_targets
    c = SkyCoord(ra=ra*u.degree, dec=dec*u.degree, frame='icrs')
  File "/home/ohsawa/.pyenv/versions/anaconda3-2020.11/lib/python3.8/site-packages/pandas/core/ops/common.py", line 65, in new_method
    return method(self, other)
  File "/home/ohsawa/.pyenv/versions/anaconda3-2020.11/lib/python3.8/site-packages/pandas/core/ops/__init__.py", line 343, in wrapper
    result = arithmetic_op(lvalues, rvalues, op)
  File "/home/ohsawa/.pyenv/versions/anaconda3-2020.11/lib/python3.8/site-packages/pandas/core/ops/array_ops.py", line 190, in arithmetic_op
    res_values = na_arithmetic_op(lvalues, rvalues, op)
  File "/home/ohsawa/.pyenv/versions/anaconda3-2020.11/lib/python3.8/site-packages/pandas/core/ops/array_ops.py", line 143, in na_arithmetic_op
    result = expressions.evaluate(op, left, right)
  File "/home/ohsawa/.pyenv/versions/anaconda3-2020.11/lib/python3.8/site-packages/pandas/core/computation/expressions.py", line 233, in evaluate
    return _evaluate(op, op_str, a, b)  # type: ignore
  File "/home/ohsawa/.pyenv/versions/anaconda3-2020.11/lib/python3.8/site-packages/pandas/core/computation/expressions.py", line 109, in _evaluate_numexpr
    result = ne.evaluate(
  File "/home/ohsawa/.pyenv/versions/anaconda3-2020.11/lib/python3.8/site-packages/numexpr/necompiler.py", line 821, in evaluate
    signature = [(name, getType(arg)) for (name, arg) in
  File "/home/ohsawa/.pyenv/versions/anaconda3-2020.11/lib/python3.8/site-packages/numexpr/necompiler.py", line 821, in <listcomp>
    signature = [(name, getType(arg)) for (name, arg) in
  File "/home/ohsawa/.pyenv/versions/anaconda3-2020.11/lib/python3.8/site-packages/numexpr/necompiler.py", line 703, in getType
    raise ValueError("unknown type %s" % a.dtype.name)
ValueError: unknown type object
```

**Comment by HajimeKawahara** [comment]
@xr0038 
Thanks. This error looks occurs when converting units. So, I changed the type of ra and dec.
Also, added a test just for read_catalog, namely, tests/mapping/test_read_catalog.py
Can you try again?

File: src/telescope_baseline/mapping/__init__.py [hunk 1]
```diff
@@ -0,0 +1,8 @@
+__all__ = []
+
+__version__ = '0.1'
+__uri__ = ''
+__author__ = 'JASMINE e2e group'
+__email__ = ''
+__license__ = ''
+__description__ = ''
```
Comment (xr0038) [inline src/telescope_baseline/mapping/__init__.py:8]: Should we define these items in each subdirectory?
Defining them on the top level seems sufficient.

Comment (HajimeKawahara) [reply src/telescope_baseline/mapping/__init__.py]: thanks, deleted

**Comment by HajimeKawahara** [comment]
Here is a cheat sheet of aperture.py
IMG_20220509_095011

IMG_20220509_095018

File: src/telescope_baseline/mapping/aperture.py [hunk 2]
```diff
@@ -0,0 +1,293 @@
...
+       x: vectors whose cos angle from normal vector of the ABO plane to be computed 3 or (M, 3)
+     
+    Returns:
+       cos_angles, float or (N,M)
+    """
+    return (x@np.cross(a,b).T).T
+
+
+def convex_on_sphere(angv,angw):
+    """checking if the points
```
Comment (xr0038) [inline src/telescope_baseline/mapping/aperture.py:147]: Is the description is not completed?

File: src/telescope_baseline/mapping/aperture.py [hunk 3]
```diff
@@ -0,0 +1,293 @@
...
+       b: 3D vector of B, (3,) or (N, 3)
+       x: vectors whose cos angle from normal vector of the ABO plane to be computed 3 or (M, 3)
+     
+    Returns:
+       cos_angles, float or (N,M)
+    """
+    return (x@np.cross(a,b).T).T
+
+
+def convex_on_sphere(angv,angw):
```
Comment (xr0038) [inline src/telescope_baseline/mapping/aperture.py:146]: The function name does not represent its functionality.
Reconsider the function name.

**Comment by xr0038** [comment]
@HajimeKawahara 
Thank you! I confirmed that the code passes the test and the example works properly.
I continue to review the code.

File: src/telescope_baseline/mapping/aperture.py (hunk 3 above)
Comment (HajimeKawahara) [reply src/telescope_baseline/mapping/aperture.py]: right. I changed

Comment (xr0038) [reply src/telescope_baseline/mapping/aperture.py]: thx!

**Pull request merged**